lib/cli/handle-error.js

```diff
@@ -41,6 +45,7 @@ module.exports = async (exception, options = {}) => {
     isLocallyInstalled: passedIsLocallyInstalled,
     isInvokedByGlobalInstallation: passedIsInvokedByGlobalInstallation,
     serverless,
+    hasTelemetryBeenReported,
```

Comment:
Here I've decided that we should disregard situations where the handler is invoked from older global version as a fallback and still report the failed command vs not doing it. Please let me know if you feel otherwise

Reply:
That's a good point, as I didn't cover the scenario of fallback in spec.

Technically sending telemetry unconditionally may result with doubled telemetry for same command use, as in old version we report telemetry before command starts (if it errors it's most likely after telemetry was send). So I'm on a fence here.

To fix that it'll probably be better to send telemetry here only if `hasTelemetryBeenReported === false`, that'll ensure there's no double records, and that there's a clear division between versions that report `outcome` in all cases, and those which do not at all

Reply:
My reasoning here was that in such situations, we still get useful information as the potential "double" telemetry will be a bit different. The first even won't have the `outcome` property, so we don't know if it succeeded or failed while the second one will have the `outcome: failure`. I'm fine however with not sending telemetry in such case.

Reply:
I thought at first glance as well, that it's probably useful, but when I look more deeply, I get following thoughts

- Programmer error reports will be useless, as we should focus just on those coming from latest versions (old version will report errors we already fixed, and if not those will be reported by latest versions as well)
- Info on general failures will be incomplete. As they will be send just in case of local fallback from latest global, and it's hard to estimate what percentage that is, which means we cannot reliably state how many % of failures we have there, and I assume we will rather avoid confirming on error rates in those verions (we will prefer to exclude them from stats completely)
- When we will want to narrow in mixpanel output to just versions that consistently report `outcome`, we will have to again rely on specific Framework version. While if we decide to not report errors in such case, we can safely just estimate on existence of `outcome` property.

What are your thoughts on that?

Reply:
I think that makes sense, especially the last part:

> When we will want to narrow in mixpanel output to just versions that consistently report outcome, we will have to again rely on specific Framework version. While if we decide to not report errors in such case, we can safely just estimate on existence of outcome property.

which makes reviewing such errors a lot more convenient :+1:

Reply:
changed :+1: